File: src/main/java/com/puertobahia/iceberg/service/ActividadService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.puertobahia.iceberg.service;


import com.puertobahia.iceberg.entity.Actividad;
import java.util.List;

/**
 *
 * @author Full Stack JavaScrip
 */
public interface ActividadService {
    
    public List<Actividad> getAllActividad();
    public Actividad getById(Long id);
    public void saveOrUpdate(Actividad actividad);
    public void delete(Long id);
    
    
}

File: src/main/webapp/resources/assets/js/app/services/actividadService.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

var servicesModule = angular.module('AppServices');

servicesModule.factory('actividadService', ['$http', function ($http) {
    return {
        apiUrl: apiUrl,
        getAllActividad: function () {
            return $http.get(this.apiUrl + 'actividad/');
        },
        getActividadById: function (actividadId) {
            return $http.get(this.apiUrl + 'actividad/' + actividadId);
        },
        createActividad: function (actividad) {
            return $http.post(this.apiUrl + 'actividad/', actividad);
        },
        updateActividad: function (actividad) {
            return $http.put(this.apiUrl + 'actividad/', actividad);
        },
        deleteActividad: function (actividadId) {
            return $http.delete(this.apiUrl + 'actividad/' + actividadId);
        }
    };
    }]);
